### PR TITLE
Convert patients/search/NoPatientsExist.test.tsx to RTL

### PR DESCRIPTION
**Pull request by Proful**
Fixes #30

### CODE REVIEW

File: src/__tests__/patients/search/NoPatientsExist.test.tsx [hunk 1]
```diff
@@ -1,33 +1,18 @@
...
-    expect(addNewPatient).toHaveLength(1)
+    expect(screen.getByRole('heading')).toBeInTheDocument()
+    expect(screen.getByText('patients.noPatients')).toBeInTheDocument()
 
-    const icon = wrapper.find(Icon).first()
-    const typography = wrapper.find(Typography)
-    const button = wrapper.find(Button)
-    const iconType = icon.prop('icon')
-    const iconSize = icon.prop('size')
-    const typographyText = typography.prop('children')
```
Comment (Proful) [inline src/__tests__/patients/search/NoPatientsExist.test.tsx:21]: Should we test icon/size These are not rendered in actual DOM

File: src/__tests__/patients/search/NoPatientsExist.test.tsx [hunk 2]
```diff
@@ -1,33 +1,18 @@
...
-    const typographyVariant = typography.prop('variant')
-    const buttonIcon = button.prop('icon')
-    const buttonText = button.prop('children')
-
-    expect(iconType).toEqual('patients')
-    expect(iconSize).toEqual('6x')
-    expect(typographyText).toEqual('patients.noPatients')
-    expect(typographyVariant).toEqual('h5')
```
Comment (Proful) [inline src/__tests__/patients/search/NoPatientsExist.test.tsx:29]: h5 is implementation detail isn't it?
If someone changed it to h6 or h4 do we want to break test?

Comment (nobrayner) [reply src/__tests__/patients/search/NoPatientsExist.test.tsx:29]: I'd try using a `getByRole('heading', { name: /xxx/i })`

Comment (Proful) [reply src/__tests__/patients/search/NoPatientsExist.test.tsx:29]: Added `expect(screen.getByRole('heading', { name: /patients.noPatients/i })).toBeInTheDocument()`

File: src/__tests__/patients/search/NoPatientsExist.test.tsx [hunk 3]
```diff
@@ -1,33 +1,18 @@
...
-    expect(buttonIcon).toEqual('patient-add')
-    expect(buttonText).toEqual('patients.newPatient')
+    expect(screen.getByRole('button')).toBeInTheDocument()
+    expect(screen.getByText('patients.newPatient')).toBeInTheDocument()
```
Comment (nobrayner) [inline src/__tests__/patients/search/NoPatientsExist.test.tsx:16]: This looks like you are no-longer checking for the icon - was there a reason for this? You can probably also combine the two assertions by specifying the buttons name

**Comment by Proful** [comment]
@nobrayner is this PR looks good? ready to merge?

File: src/__tests__/patients/search/NoPatientsExist.test.tsx [hunk 4]
```diff
@@ -1,33 +1,18 @@
...
+  const setup = () => render(<NoPatientsExist />)
 
   it('should render an icon and a button with typography', () => {
-    const wrapper = setup()
+    setup()
 
-    const addNewPatient = wrapper.find(NoPatientsExist)
-    expect(addNewPatient).toHaveLength(1)
+    expect(screen.getByRole('heading', { name: /patients.noPatients/i })).toBeInTheDocument()
+    expect(screen.getByText('patients.noPatients')).toBeInTheDocument()
```
Comment (nobrayner) [inline src/__tests__/patients/search/NoPatientsExist.test.tsx:13]: These two lines appear to be the checking the same thing... Can you collapse them?
```suggestion
    expect(screen.getByRole('heading', { name: /patients.noPatients/i })).toBeInTheDocument()
```

**Review by nobrayner — COMMENTED** [review]
See comments.

**Comment by Proful** [comment]
Some one else raised a PR. Closing this.

**Pull request closed without merging**